prelude.jq

```diff
@@ -91,3 +91,12 @@ def todateiso8601: strftime("%Y-%m-%dT%H:%M:%SZ");
 def fromdate: fromdateiso8601;
 def todate: todateiso8601;
 
+def match($r; $f): __match_impl($r; $f)[];
+def match($r): if $r | type == "array" then match($r[0]; $r[1]) else match($r; null) end;
+def test($r; $f): isempty(match($r; $f)) | not;
+def test($r): if $r | type == "array" then test($r[0]; $r[1]) else test($r; null) end;
+def capture($r; $f): match($r; $f) | [.captures[] | select(.name) | {key: .name, value: .string}] | from_entries;
+def capture($r): if $r | type == "array" then capture($r[0]; $r[1]) else capture($r; null) end;
+def scan($r; $f): match($r; $f + "g") | .captures | [.[].string];
```

Comment:
Better or worse, jq emits the matched string if there is no captures.
```sh
❯ jq -n '"abc" | scan("b")'
"b"
```

Reply:
Thanks, updated to handle it!